common/flatpak-context.c

```diff
@@ -3628,6 +3629,15 @@ FlatpakContextSockets
 flatpak_context_compute_allowed_sockets (FlatpakContext                   *context,
                                          FlatpakContextConditionEvaluator  evaluator)
 {
+  /* The fallback-x11 socket permission was an ad-hoc fallback mechanism which
+   * we can take care of here so the caller can just ignore it. */
+  if (context->sockets & FLATPAK_CONTEXT_SOCKET_FALLBACK_X11)
+    {
+      context->sockets &= ~FLATPAK_CONTEXT_SOCKET_FALLBACK_X11;
+      if (!evaluator || evaluator (FLATPAK_CONTEXT_CONDITION_HAS_WAYLAND) == FALSE)
+        context->sockets |= FLATPAK_CONTEXT_SOCKET_X11;
+    }
+
```

Comment:
Is it really good to modify the context like this? If we then re-serialize the context we will lose the fallback-x11 option. I'm not sure exactly if this would happen, but it seems risky.

Reply:
True, and there is no reason to modify the context. This now acts on the sockets we pass to `flatpak_context_compute_allowed`, not on the context.